### ISSUE TITLE
Failing PHP CS Fixer Tool Tests 

### ISSUE THREAD

**Issue by 36degrees** [markstory/lint-review#67]
Hi,

Trying to boot `lint-review` using `docker-compose` and I'm seeing the following:

```
======================================================================
FAIL: test_process_files__one_file_fail (tests.tools.test_phpcs.Testphpcs)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/code/tests/tools/test_phpcs.py", line 45, in test_process_files__one_file_fail
    eq_(12, len(problems))
AssertionError: 12 != 3
-------------------- >> begin captured logging << --------------------
lintreview.tools.phpcs: DEBUG: Processing ['tests/fixtures/phpcs/has_errors.php'] files with phpcs
lintreview.tools: DEBUG: Running phpcs --report=checkstyle --standard=PSR2 --extensions=php tests/fixtures/phpcs/has_errors.php
lintreview.review: DEBUG: Adding error '<class 'lintreview.review.Comment'>(filename=tests/fixtures/phpcs/has_errors.php, line=14, position=14, body=Opening brace should be on a new line)'
lintreview.review: DEBUG: Adding error '<class 'lintreview.review.Comment'>(filename=tests/fixtures/phpcs/has_errors.php, line=15, position=15, body=Spaces must be used to indent lines; tabs are not allowed)'
lintreview.review: DEBUG: Adding error '<class 'lintreview.review.Comment'>(filename=tests/fixtures/phpcs/has_errors.php, line=16, position=16, body=Spaces must be used to indent lines; tabs are not allowed)'
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: test_process_files_two_files (tests.tools.test_phpcs.Testphpcs)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/code/tests/tools/test_phpcs.py", line 65, in test_process_files_two_files
    eq_(12, len(problems))
AssertionError: 12 != 3
-------------------- >> begin captured logging << --------------------
lintreview.tools.phpcs: DEBUG: Processing ['tests/fixtures/phpcs/no_errors.php', 'tests/fixtures/phpcs/has_errors.php'] files with phpcs
lintreview.tools: DEBUG: Running phpcs --report=checkstyle --standard=PSR2 --extensions=php tests/fixtures/phpcs/no_errors.php tests/fixtures/phpcs/has_errors.php
lintreview.review: DEBUG: Adding error '<class 'lintreview.review.Comment'>(filename=tests/fixtures/phpcs/has_errors.php, line=14, position=14, body=Opening brace should be on a new line)'
lintreview.review: DEBUG: Adding error '<class 'lintreview.review.Comment'>(filename=tests/fixtures/phpcs/has_errors.php, line=15, position=15, body=Spaces must be used to indent lines; tabs are not allowed)'
lintreview.review: DEBUG: Adding error '<class 'lintreview.review.Comment'>(filename=tests/fixtures/phpcs/has_errors.php, line=16, position=16, body=Spaces must be used to indent lines; tabs are not allowed)'
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 136 tests in 42.757s

FAILED (SKIP=9, failures=2)
Service 'image' failed to build: The command '/bin/sh -c nosetests -v' returned a non-zero code: 1
```

It looks as though it's expecting PHP CS to find 12 errors in the file, but it's only getting 3. Running the tool manually (using PHP_CodeSniffer version 2.3.3) locally returns 3 as well:

```
phpcs --report=checkstyle --standard=PSR2 tests/fixtures/phpcs/has_errors.php
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle version="2.3.3">
<file name="/Users/ollie/Code/lint-review/tests/fixtures/phpcs/has_errors.php">
 <error line="14" column="36" severity="error" message="Opening brace should be on a new line" source="Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine"/>
 <error line="15" column="1" severity="error" message="Spaces must be used to indent lines; tabs are not allowed" source="Generic.WhiteSpace.DisallowTabIndent.TabsUsed"/>
 <error line="16" column="1" severity="error" message="Spaces must be used to indent lines; tabs are not allowed" source="Generic.WhiteSpace.DisallowTabIndent.TabsUsed"/>
</file>
</checkstyle>
```

The composer.lock file is tied to PHPCS 2.2.0 (https://github.com/squizlabs/PHP_CodeSniffer/commit/b301c98f19414d836fdaa678648745fcca5aeb4f) but it looks like PEAR installs 2.3.3:

```
Step 6 : RUN pear install PHP_CodeSniffer
 ---> Running in d8adf72d1c51
downloading PHP_CodeSniffer-2.3.3.tgz ...
Starting to download PHP_CodeSniffer-2.3.3.tgz (470,589 bytes)
...............................................................................................done: 470,589 bytes
install ok: channel://pear.php.net/PHP_CodeSniffer-2.3.3
```

Presumably something's changed between 2.2.0 and 2.3.3 that means less errors are being reported?

Regardless, this causes the boot to fail and makes it unusable without a lot of faff.

Ta

Ollie

**Comment by 36degrees** [comment]
FWIW, I've tried changing the call to pear in the Dockerfile to `RUN pear install PHP_CodeSniffer-2.2.0`, but it still appears to fail in exactly the same way.

**Comment by markstory** [comment]
I bet phpcs has changed the lint rules it uses since the tests were written. The code still works, bit I will fix the tests.

**Comment by 36degrees** [comment]
Thanks Mark.

The weird bit is the fact that installing 2.2.0 doesn't fix it – I'd have assumed that the rules wouldn't change within the same version?

FYI, I'm finding a number of things in the readme that aren't right / applicable when using docker (like the use of `activate`) – will try and put a PR in to update it if I get a chance. :smile:

**Comment by markstory** [comment]
The failing tests have been fixed in 26a3f73ac1039b4014d9b0627f652f846c4c0adb

**Comment by 36degrees** [comment]
Thanks @markstory!
